src/quo/components/wallet/transaction_progress/view.cljs

```diff
@@ -0,0 +1,178 @@
+(ns quo.components.wallet.transaction-progress.view
+  (:require [quo.components.buttons.button.view :as button]
+            [quo.components.icon :as icons]
+            [quo.components.markdown.text :as text]
+            [quo.components.tags.context-tag.view :as context-tag]
+            [quo.components.wallet.confirmation-progress.view :as confirmation-progress]
+            [quo.components.wallet.transaction-progress.style :as style]
+            [quo.foundations.colors :as colors]
+            [quo.theme :as quo.theme]
+            [react-native.core :as rn]
+            [utils.i18n :as i18n]))
+
+(defn- icon-internal
+  ([icon]
+   (icon-internal icon nil 20))
+  ([icon color]
+   (icon-internal icon color 20))
+  ([icon color size]
+   [rn/view {:style style/icon}
+    [icons/icon
+     icon
+     {:color    color
+      :size     size
+      :no-color (when (not color) true)}]]))
+
+(defn- text-internal
+  [{:keys [weight size accessibility-label color]
+    :or   {weight :semi-bold
+           size   :paragraph-1}}
+   title]
+  [text/text
+   {:accessibility-label accessibility-label
+    :ellipsize-mode      :tail
+    :number-of-lines     1
+    :weight              weight
+    :size                size
+    :style               {:color color}}
+   title])
+
+(defn- network-type-text
+  [state]
+  (cond
+    (or (= state :sending) (= state :pending))      (i18n/label :t/pending-on)
+    (or (= state :confirmed) (= state :finalising)) (i18n/label :t/confirmed-on)
+    (= state :finalized)                            (i18n/label :t/finalized-on)
+    (= state :error)                                (i18n/label :t/failed-on)))
+
+(defn- calculate-counter
+  [counter]
```

Comment:
it's better to keep functions like this pure and pass in the de ref'd value.
i.e
```
(defn- calculate-counter
  [counter]
(if (< counter 4) counter "4"))
```

```
(calculate-counter @some-counter)
```

Reply:
Done